src/components/shared/ContractFunction.vue

```diff
@@ -0,0 +1,366 @@
+<script setup lang="ts">
+import useWeb3 from '@/compositions/useWeb3'
+import { BN } from 'bn.js'
+import { reactive, computed, watch, onMounted } from 'vue'
+import { toWei, Unit } from 'web3-utils'
+import { MethodSelect, MethodType } from '../endpoints/SendTransaction.vue'
+import ParamField from './ParamField.vue'
+
+interface ElementType extends MethodType {
+  error?: boolean
+}
+type Props = {
+  call?: string
+  modelValue?: MethodType[]
+  custom?: boolean
+  onlyParams?: boolean
+  hideData?: boolean
+  data?: string
+  testidPrefix?: string
+}
+
+type Emits = {
+  (event: 'update:modelValue', value: any): void
+  (event: 'update:error', value: boolean): void
+  (event: 'update:data', value: string): void
+}
+
+const props = defineProps<Props>()
+const emits = defineEmits<Emits>()
+
+const { getWeb3 } = useWeb3()
+const { eth } = getWeb3()
+
+const SIGNATURE_CACHE = 'signature-cache'
+
+type BytesSignatureResponse = {
+  count: number
+  next: unknown
+  previous: unknown
+  results: [
+    {
+      id: number
+      created_at: string
+      text_signature: string
+      hex_signature: string
+      bytes_signature: string
+    }
+  ]
+}
+
+const fetcher = async <Response, Request>(config: {
```

Comment:
Let's move `fetcher` function out of the component to some util

Reply:
Moving